### DISCUSSION TITLE
False person detection on immobile object in latest dev

### DISCUSSION THREAD

**Discussion by 0anton** [blakeblackshear/frigate#8669] [Ask A Question]
Dear community,

I'm strugling to lower the rate of false positives for people detection.
Below is the general and the detailed view. As you can see the covered folowers are detected as a person with high confidence.

I find it is strange, because:
- the object is not moving
- there is no wind
-  camera is firmly mounted on the wall of the building

In my understanding the object must not trigger the motion detection.

I've checked the image in the debug mode - nothing suspicious in the area - motion boxes, bounding boxes are not visible at the time of the check.

I've increased detection threshold to `0.75` (as shown below)

I cannot apply max/min size limits, because this area is exactly the size the person can be.

I've moved to latest development snapshot [1] in the hope that improved motion detection will help to filter the object.

```
objects:
  track:
    - person
  filters:
    person:
      min_score: 0.5
      threshold: 0.75
```

What can I try to decrease the false positives of that kind?
I'd glad to try other option combinations.

**Refs**
1. https://ghcr.io/blakeblackshear/frigate:dev-7d157df

![image](https://github.com/blakeblackshear/frigate/assets/17309924/7f5b1596-d58f-4e16-ad0a-faf8a2d99dbb)
![image](https://github.com/blakeblackshear/frigate/assets/17309924/6b1c1cc0-d31b-4053-8d02-eab8022acabe)
![photo_2023-11-19_15-29-48](https://github.com/blakeblackshear/frigate/assets/17309924/5a50895e-5c65-424d-bfda-f16bce9cf9bc)
![image](https://github.com/blakeblackshear/frigate/assets/17309924/3175c151-8a46-467d-a38b-eade66a5fda4)

**Comment by NickM-27** [comment]
After frigate is started and running normally, motion detection is required for object detection, once an object is detected it will be tracked continuously. Most likely some bit of motion started the detection and is no longer occurring. 

Given that this object is static and in similar position I think an object mask may be appropriate here

**Comment by 0anton** [comment]
> Given that this object is static and in similar position I think an object mask may be appropriate here

My parents can create more of such objects in the survivalence area 🙂 Masking such areas does not sounds like a constructive approach as it will hinder detection of real objects.

Why in the first place a pile of branches on the ground is detected as a human with 78% confidence? I don't see even a high-level similarity...

**Reply by NickM-27** [reply]
The default model in frigate is a Google demo model trained on the COCO dataset. The COCO dataset does not contain images that look like security camera perspective, so this leads to false positives. Here are some examples from the COCO dataset

![pizza](https://i.imgur.com/drRR7dn.png)
![reflection](https://i.imgur.com/uhsuDdp.png)
![hand](https://i.imgur.com/Nb3B8RM.png)

**Reply by 0anton** [reply]
I see the point. I understand now the importance of your workstream to add model feedback option and allow custom / fune-tuned models.

**Comment by 0anton** [comment]
Hi Nick,

I understand it very well. But I think the problem is more on the motion detection side than on image recognition. A lot of false positives will disappear if the motion detection will be more accurate.

Can for example this static object not be put in to te object detection pipeline? Stationary object (water pipe), non-moving camera...

![photo_2023-11-23_12-07-28](https://github.com/blakeblackshear/frigate/assets/17309924/f87c6547-578e-4bff-a862-e016e790f694)
![photo_2023-11-23_12-07-27](https://github.com/blakeblackshear/frigate/assets/17309924/f9e535e2-857f-4b29-8429-664f356756aa)
![photo_2023-11-23_12-07-25](https://github.com/blakeblackshear/frigate/assets/17309924/28d07eff-097f-4a47-bc6a-303183dca6ca)
![photo_2023-11-23_12-07-24](https://github.com/blakeblackshear/frigate/assets/17309924/f639d195-c78d-4105-8d60-e57036578e60)
![photo_2023-11-23_12-07-22](https://github.com/blakeblackshear/frigate/assets/17309924/54de4e0a-8b58-4548-8838-0d2a557d3c14)
![photo_2023-11-23_12-07-21](https://github.com/blakeblackshear/frigate/assets/17309924/b82b4aef-dab1-4fea-b165-f5e99bea395a)
![photo_2023-11-23_12-07-19](https://github.com/blakeblackshear/frigate/assets/17309924/07b83404-9808-4e25-b6a6-c015b6801051)

**Reply by NickM-27** [reply]
You should tune your motion detection according to the docs https://deploy-preview-6262--frigate-docs.netlify.app/configuration/motion_detection

**Comment by substeeze** [comment]
Old thread, but in case it helps anyone else... I was experiencing a similar issue: once motion detection had kicked off object recognition, it just kept seeing a 'person' and was constantly running the object detection process.

An object mask will prevent an alert being triggered, but doesn't stop the object recognition process running.

I fixed this problem by tweaking my camera settings. For some reason, disabling '3D Noise Reduction' on my Reolink camera configuration page seemed to work. Nothing to do with Frigate settings at all, but something in the noise reduction on the camera that was upsetting the detector in Frigate.
